docs/high-availability/provision.sh

```diff
@@ -0,0 +1,179 @@
+        sed "s/NODE_IP/$IP/" > /etc/kubernetes/manifests/etcd.manifest
+}
+
+### Test of ETCD Members.
+
+function test_etcd {
+    
+    echo "----------- DEBUG ------------ KUBELET LOGS -----------------"
+    sudo journalctl -u kubelet | grep -A 20 -B 20 Fail
+    echo "----------- END DEBUG OF KUBELET ----------------------------"
+
+    curl http://kube0.ha:2379 > /tmp/curl_output
+    if [ -s /tmp/curl_output ]; then
+         echo "etcd success" 
+    else
+         echo "etcd failure.  exit!"
+         exit 100 
+    fi
+   
+}
+
+function k8petstore {
+     ### run K8petstore .  Should work perfectly IFF flannel and so on is setup properly.
+    wget https://raw.githubusercontent.com/GoogleCloudPlatform/kubernetes/release-0.17/examples/k8petstore/k8petstore.sh
+    chmod 777 k8petstore.sh
+    ./k8petstore.sh
+}
+
+function write_api_server_config { 
+    touch /var/log/kube-apiserver.log
+    mkdir -p -m 777 /srv/kubernetes/    
+    
+    ### We will move files back and forth between the /srv/kube.. directory.
+    ### That is how we modulate leader.  Each node will continously either 
+    ### ensure that the manifests are in this dir, or else, are in the kubelet manifest dir.
+    cp /vagrant/kube-scheduler.manifest /vagrant/kube-controller-manager.manifest /srv/kubernetes
+    
+    ### All nodes will run an API Server.  This is because API Server is stateless, so its not a problem
+    ### To serve it up everywhere.
+    cp /vagrant/kube-apiserver.manifest /etc/kubernetes/manifests/
+}
+
+function write_podmaster_config {
+    touch /var/log/kube-scheduler.log
+    touch /var/log/kube-controller-manager.log
+ 
+    ### These DO NOT go in manifest. Instead, we mount them here.
+    ### We let podmaster swap these in and out of the manifests directory
+    ### based on its own internal HA logic.
+    cp /vagrant/kube-controller-manager.manifest /srv/kubernetes/
+    cp /vagrant/kube-scheduler.manifest /srv/kubernetes/
+
+    #### Finally, the podmaster is the mechanism for election
+    cp /vagrant/podmaster.json /etc/kubernetes/manifests/
+}
+
+function poll {
+    ### wait 10 minutes for kube-apiserver to come online
+    for i in `seq 1 600`;
+       do
+           sleep 2
+           ### Just testing that the front end comes up.  Not sure how to test total entries etc... (yet)
+           echo "Trying curl ... $i . expect a few failures while pulling kube-apiserver images and so on ... " 
+           curl "localhost:8080" > result
+           cat result
+           cat result | grep -q "api"
+           if [ $? -eq 0 ]; then
+              echo "TEST PASSED after $i tries !"
+              i=1000
+              break
+           else    
+              echo "the above RESULT didn't contain target string for trial $i"
+           fi
+           if [ $i -eq 1000 ]; then
+               echo "EXITING ! API Server still not up after several tries :("
+               exit 2                
+           fi
+       done
```

Comment:
nit: `do` should be on the same line as `for` and `done` should have the same indentation as `for`